cmake/MiscCheck.cmake

```diff
@@ -0,0 +1,50 @@
+include(CheckCXXSourceCompiles)
+include(CMakePushCheckState)
+
+# We use the [[nodiscard]] attribute, which GCC 5 complains about.
+# Silence this warning if GCC 5 is used.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6)
+    add_definitions("-Wno-attributes")
+  endif()
+endif()
+
+if(LINUX AND TP_ENABLE_CMA)
+  # Check libc contains process_vm_readv
+  CMAKE_PUSH_CHECK_STATE(RESET)
+  set(CMAKE_REQUIRED_FLAGS "${CMAKE_CXX_FLAGS}")
+  CHECK_CXX_SOURCE_COMPILES("
+    #include <sys/uio.h>
+    #include <sys/types.h>
+    #include <unistd.h>
+
+    namespace {
+    bool isProcessVmReadvSyscallAllowed() {
+      long long someSourceValue = 0x0123456789abcdef;
+      long long someTargetValue = 0;
+      struct iovec source = {
+        .iov_base = &someSourceValue, .iov_len = sizeof(someSourceValue)
+      };
+      struct iovec target = {
+        .iov_base = &someTargetValue, .iov_len = sizeof(someTargetValue)
+      };
+      ssize_t nread = ::process_vm_readv(::getpid(), &target, 1, &source, 1, 0);
+      return nread == sizeof(long long) && someTargetValue == someSourceValue;
+    }
+    }
```

Comment:
This code is just required to compile right? It doesn't have to be be correct when executed? In that case, we can probably simplify this simply to:
```
int main() {
  ssize_t rv = ::process_vm_readv(1, nullptr, 0, nullptr, 0, 0);
  return 0;
}
```
Plus possibly casting arguments to the right types for extra safety.

Reply:
yes. that's a good point. we should simplify, sorry for just copy-pasting the code from context_impl.cc